src/App.tsx

```diff
@@ -1,61 +1,117 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/types';
+
+const initialTodos: Todo[] = todosFromServer;
+
+function getNewTodoId(todos: Todo[]) {
+  const maxId = Math.max(...todos.map(todo => todo.id));
+
+  return maxId + 1;
+}
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+
+  const addTodo = ({ id, ...data }: Todo) => {
+    const newTodo = {
+      id: getNewTodoId(todos),
+      ...data,
+    };
+
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  function handleTitle(e: React.ChangeEvent<HTMLInputElement>) {
+    setTitle(e.target.value);
+    setHasTitleError(false);
+  }
+
+  function handleUserId(e: React.ChangeEvent<HTMLSelectElement>) {
+    setUserId(+e.target.value);
+    setHasUserIdError(false);
+  }
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const newTodoObj = {
+      id: 0,
+      title: title.trim(),
+      userId,
+      completed: false,
+    };
+
+    addTodo(newTodoObj);
+    setTitle('');
+    setUserId(0);
+  }
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="input-title">Title: </label>
+          <input
+            id="input-title"
+            type="text"
+            data-cy="titleInput"
+            placeholder="Please enter a title"
+            value={title}
+            onChange={handleTitle}
+          />
+          {hasTitleError && (
+            <span className="error">Please enter a title{title}</span>
```

Comment:
The error message for the title input should not include the title itself. It should simply be 'Please enter a title'.